platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ResolutionBackedFileCollection.java

```diff
@@ -77,7 +77,7 @@ protected void visitContents(FileCollectionStructureVisitor visitor) {
     /**
      * If the file collection is not lenient, rethrow any failures that occurred during the visit.
      *
-     * @throws ArtifactVariantSelectionException subtypes
+     * @throws ArtifactSelectionException subtypes
```

Comment:
I don't think we can be specific here. The thrown error _may_ be one of these exceptions, but I think we can get others, like failures that occur when _downloading_ artifacts.